### PR TITLE
API changes + code cleanup

### PR DESCRIPTION
**Pull request by mehmetf**

### CODE REVIEW

**Comment by mehmetf** [comment]
@Lyokone Please review this change.

- Many of the files are due to updates flutter has been doing to the skeleton project. Without these flutter run does not run the sample app.
- A couple of edits to the Android and iOS part of the plugin.
 - on iOS, we can't assume that the build system is based on CocoaPods so we check for its existence.
 - on Android, LocationRequest ctr is now private. Instead we need to use `create()` method.
- Returning a Map is much less safe than returning an object. You are not leaking your const's to the app.

File: example/ios/Flutter/AppFrameworkInfo.plist [hunk 1]
```diff
@@ -2,33 +2,25 @@
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
-	<key>NSLocationAlwaysUsageDescription</key>
```
Comment (khmoran) [inline example/ios/Flutter/AppFrameworkInfo.plist]: Do we need these the two location keys? 

```
NSLocationAlwaysUsageDescription
NSLocationWhenInUseUsageDescription
```

Comment (mehmetf) [reply example/ios/Flutter/AppFrameworkInfo.plist]: I had not tried it on iOS yet. Thanks for catching this!

**Comment by mehmetf** [comment]
Since this is a blocker for bringing the plugin to Google, I am going to submit these changes.

@Lyokone, if you are not comfortable with anything in this PR, please comment on it and I will do my best to accommodate.  The version has been upped to 2.0.0 to indicate a breaking change.

**Comment by mehmetf** [comment]
Also note that we don't consume code from pub so it is entirely up to you when and how to publish these versions to pub.

**Comment by Lyokone** [comment]
> Since this is a blocker for bringing the plugin to Google, I am going to submit these changes.
> 
> @Lyokone, if you are not comfortable with anything in this PR, please comment on it and I will do my best to accommodate. The version has been upped to 2.0.0 to indicate a breaking change.

The changes seems good on my end. I'm probably not on the same timezone as you so don't hesitate to merge even if I haven't approved the PR.

I'll try to keep the pub version updated.

**Pull request merged**